src/EditorFeatures/CSharpTest/DisposeAnalysis/DisposeObjectsBeforeLosingScopeTests.cs

```diff
@@ -5041,5 +5041,24 @@ void M1()
 }",
             Diagnostic(IDEDiagnosticIds.DisposeObjectsBeforeLosingScopeDiagnosticId));
         }
+
+        [Fact, WorkItem(36498, "https://github.com/dotnet/roslyn/issues/36498")]
+        public async Task DisposableObjectPushedToStackIsNotFlagged()
+        {
+            await TestDiagnosticMissingAsync(@"
+using System;
+using System.Collections.Generic;
+
+class C : IDisposable
+{
+    public void Dispose() { }
+
+    public void M1(Stack<object> stack)
+    {
+        var c = [|new C()|];
```

Comment:
why only stack? or is it anything that pushed to collection?

Reply:
Ah, the test is basically for the underlying repro provided. We basically bail out for any method invocation in metadata where disposable object is passed as argument. We don't know if the method is adding it to collection, some factory registration method, etc. which is taking up dispose ownership. This scopes down the analysis for such cases.